### PR TITLE
[Gluten-846] Add new doc for S3 related configs and use cases of Gluten. 

### PR DESCRIPTION
**Pull request by akashsha1**
## What changes were proposed in this pull request?

Adding some documentation for Gluten use cases around S3. 

## How was this patch tested?

Not Applicable. Doc change.

### CODE REVIEW

**Comment by zhouyuan** [comment]
@akashsha1 can you please take a look on this patch? https://github.com/oap-project/gluten/pull/632

**Review by FelixYBW — APPROVED** [review]

File: docs/UsingS3.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Gluten is not able to reuse older caches for now, and the old cache files are left there after Spark context shutdown.
+
+Running spark in standalone mode
+
+If you are running Spark in standalone mode, please make sure that the gluten jars path is part of the following configs in spark-defaults.conf:
+spark.driver.extraClassPath
+spark.executor.extraClassPath
+
+For example:
+spark.driver.extraClassPath        /home/ubuntu/spark/jars/jars_ext/aws-java-sdk-bundle-1.11.271.jar:/home/ubuntu/gluten_jars/gluten-spark3.2_2.12-1.0.0-SNAPSHOT-jar-with-dependencies.jar
```
Comment (FelixYBW) [inline docs/UsingS3.md:22]: use ``` as code blog.

File: docs/UsingS3.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+spark.gluten.sql.columnar.backend.velox.cacheEnabled // enable or disable velox cache, default False
+spark.gluten.sql.columnar.backend.velox.cachePath  // the folder to store the cache files, default to /tmp
+spark.gluten.sql.columnar.backend.velox.cacheSize  // the total size of the cache, default to 128MB
+spark.gluten.sql.columnar.backend.velox.cacheShards // the shards of the cache, default to 1
+```
+
+It's recommended to mount SSDs to the cache path to get the best performance of local caching. 
+On the start up of Spark context, the cache files will be allocated under "spark.gluten.sql.columnar.backend.velox.cachePath", with UUID based suffix, e.g. "/tmp/cache.13e8ab65-3af4-46ac-8d28-ff99b2a9ec9b0". 
+Gluten is not able to reuse older caches for now, and the old cache files are left there after Spark context shutdown.
+
```
Comment (akashsha1) [inline docs/UsingS3.md:14]: I'll remove edits from the Velox.md for some of these configs, and port them here.

Comment (zhouyuan) [reply docs/UsingS3.md:19]: yes, note other than the simple auth, gluten also support the "credentials" based auth now

**Comment by zhouyuan** [comment]
note, will merge this first and Akash will follow up on the comments

**Pull request merged**